libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRouteProviderListener.java

```diff
@@ -0,0 +1,35 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.support.annotation.Nullable;
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse;
+import com.mapbox.services.android.navigation.v5.route.FasterRoute;
+import com.mapbox.services.android.navigation.v5.route.RouteListener;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
+
+import timber.log.Timber;
+
+class NavigationRouteProviderListener implements RouteListener {
```

Comment:
`NavigationRouteProviderListener` is called "provider" but doesn’t have any method that returns an object. That's confusing IMO.

Reply:
Also I'm wondering that maybe we should even remove this class completely. The functionality included is only "delegating" through the `NavigationEventDispatcher`. Couldn't `NavigationEventDispatcher` implement `RouteListener`? I think, creating a new class here is not worth it.
IMO, objects and methods should live/be where their context is.

Reply:
Decided to update to `NavigationFasterRouteListener` so we can still keep the faster route detection in the listener confined to a single class - let me know what you think 👀